control/plugin/client/grpc.go

```diff
@@ -305,6 +301,14 @@ func ToCoreMetrics(mts []*rpc.Metric) []core.Metric {
 }
 
 func ToCoreMetric(mt *rpc.Metric) core.Metric {
+	if ts := mt.Timestamp; ts == nil {
+		mt.Timestamp = ToTime(time.Now())
+	}
+
+	if lastAdvTime := mt.LastAdvertisedTime; lastAdvTime == nil {
```

Comment:
Could you simplify the conditional to be like this: 
`if mt.LastAdvertisedTime == nil {...`